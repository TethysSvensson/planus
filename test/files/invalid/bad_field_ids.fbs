table Foo {
    x: int32 (id: 0);
    y: int32;
}

table Bar {
    x: int32 (id: 1);
    y: int32 (id: 2);
}

table Baz {
    x: int32 (id: 0);
    y: int32 (id: 2);
}

table Quux {
    x: int32 (id: 0);
    y: int32 (id: 3);
    z: int32 (id: 5);
}
