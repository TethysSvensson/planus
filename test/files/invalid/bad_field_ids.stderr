error: If any field has an id attribute, all fields must have one
  ┌─ test/files/invalid/bad_field_ids.fbs:2:5
  │
2 │     x: int32 (id: 0);
  │     ^^^^^^^^^^^^^^^^^ First attribute with an id attribute was here
3 │     y: int32;
  │     ^^^^^^^^^ First attribute without an id attribute was here

error: Table contains non-consecutive ids. Missing id 0
  ┌─ test/files/invalid/bad_field_ids.fbs:6:1
  │
6 │ table Bar {
  │ ^^^^^^^^^

error: Table contains non-consecutive ids. Missing id 1
   ┌─ test/files/invalid/bad_field_ids.fbs:11:1
   │
11 │ table Baz {
   │ ^^^^^^^^^

error: Table contains non-consecutive ids. Missing ids 1..2
   ┌─ test/files/invalid/bad_field_ids.fbs:16:1
   │
16 │ table Quux {
   │ ^^^^^^^^^^

error: Table contains non-consecutive ids. Missing id 4
   ┌─ test/files/invalid/bad_field_ids.fbs:16:1
   │
16 │ table Quux {
   │ ^^^^^^^^^^

Error: Bailing because of errors
